.github/scripts/trymerge.py

```diff
@@ -310,16 +353,58 @@ def check_if_should_be_merged(pr: GitHubPR, repo: GitRepo) -> None:
             print(f"Skipping rule {rule_name} due to non-matching files: {non_matching_files}")
             continue
         print(f"Matched rule {rule_name} for {pr.pr_num}")
-        return
+        return rule
     raise RuntimeError(f"PR {pr.pr_num} does not match merge rules")
 
 
+def try_revert(repo: GitRepo, pr: GitHubPR, dry_run: bool = False) -> None:
+    def post_comment(msg: str) -> None:
+        gh_post_comment(pr.org, pr.project, pr.pr_num, msg, dry_run=dry_run)
+    if not pr.is_closed():
+        return post_comment(f"Can't revert open PR #{pr.pr_num}")
+    if not RE_REVERT_CMD.match(pr.get_comment_body()):
+        raise RuntimeError(f"Comment {pr.get_comment_body()} does not seem to be a valid revert command")
+    if pr.get_comment_editor_login() is not None:
+        return post_comment("Don't want to revert based on edited command")
+    author_association = pr.get_comment_author_association()
+    author_login = pr.get_comment_author_login()
+    if author_association != "MEMBER":
+        return post_comment(f"Will not revert as @{author_login} is not a member, but {author_association}")
+
+    find_matching_merge_rule(pr, repo)
```

Comment:
Maybe add a comment that this raises on error

Reply:
I have it on the function description, but you are right, it would not hurt here